advanced-tools/gdb.rst

```diff
@@ -324,3 +324,54 @@ auto-load safe. One way to achieve this is to add a line like the following
 to ``~/.gdbinit`` (edit the specific list of paths as appropriate)::
 
     add-auto-load-safe-path ~/devel/py3k:~/devel/py32:~/devel/py27
+
+
+gdb Tips
+========
+
+Learning to use gdb effectively improves your chances of successfully
+debugging problems with Python's internals.  The tips here aren't yet
+organized in any particular way.  Over time, as the section grows,
+hopefully useful organization will become apparent.
+
+Saving and Loading Breakpoints
+------------------------------
+
+With extended exposure to particular parts of the Python runtime, you
+might find it helpful to define a routine set of breakpoints and
+commands to execute when they are hit.  (These might be appropriate to
+a specific debugging/development task, or more generally useful.)  To
+make using multiple semi-static breakpoints easier to use, you can
+save breakpoints to a file and load them in future sessions.  For
+example, you can save your current breakpoints to the file
+``python.brk`` with this command::
+
+  save breakpoints python.brk
+
+You can edit the file to your heart's content, then load it in a later
+session::
+
+  source python.brk
+
+
+Breaking at Labels
+------------------
+
+You will most often set breakpoints at the start of functions, but
+this approach is less helpful when debugging the runtime virtual
+machine, since the main interpreter loop function,
+``_PyEval_EvalFrameDefault``, is well over 4000 lines long as of 3.12.
+Fortunately, among the `many ways to set breakpoints
+<https://sourceware.org/gdb/current/onlinedocs/gdb/Specify-Location.html>`_,
```

Comment:
That link is 404 for me, should it be this?

```suggestion
<https://sourceware.org/gdb/download/onlinedocs/gdb/Specify-Location.html>`_,
```

Reply:
Actually, I think the `/download` and `/current` bits should be deleted. (I originally created this change awhile ago. It appears the structure of the GDB doc tree changed a bit.)

https://sourceware.org/gdb/onlinedocs/gdb/Specify-Location.html

I don't see how to either create a new suggestion or edit yours.

Reply:
I've updated the suggestion.